src/query/string_opfunc.c

```diff
@@ -423,7 +423,7 @@ db_string_compare (const DB_VALUE * string1, const DB_VALUE * string2, DB_VALUE
 
 	  if (!ignore_trailing_space)
 	    {
-	      ti = (string2_category == QSTR_CHAR || string2_category == QSTR_NATIONAL_CHAR);
+	      ti = (QSTR_IS_FIXED_LENGTH (str1_type) && QSTR_IS_FIXED_LENGTH (str2_type));
```

Comment:
There is the same case for db_string_compare().
It also looks good to use QSTR_IS_FIXED_LENGTH() 
It seems to be checked overall.

Reply:
I fixed it.